editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -1985,9 +1985,19 @@ void Node3DEditorViewport::_sinput(const Ref<InputEvent> &p_event) {
 			} else {
 				const bool movement_threshold_passed = _edit.original_mouse_pos.distance_to(_edit.mouse_pos) > 8 * EDSCALE;
 
-				// enable region-select if nothing has been selected yet or multi-select (shift key) is active
-				if (selection_in_progress && movement_threshold_passed) {
-					if (get_selected_count() == 0 || clicked_wants_append) {
+				if (selection_in_progress && movement_threshold_passed && clicked.is_valid()) {
+					const HashMap<Node *, Object *> &selection = editor_selection->get_selection();
+
+					bool node_in_selection_clicked = false;
+
+					for (const KeyValue<Node *, Object *> &E : selection) {
+						if (E.key->get_instance_id() == clicked) {
+							node_in_selection_clicked = true;
+							break;
+						}
+					}
```

Comment:
I think this can be done more efficiently, instead of iterating over it, I think this should be enough: `editor_selection->is_selected(Object::cast_to<Node>(ObjectDB::get_instance(clicked)));`

Reply:
Test it. It sure is.